src/Core/Package.vala

```diff
@@ -544,14 +544,19 @@ public class AppCenterCore.Package : Object {
         var list = new Gee.ArrayList<AppStream.Release> ();
 
         var releases = component.get_releases ();
-        if (releases.length < min_releases) {
-            return list;
+        uint index = 0;
+        while (index < releases.length) {
+            if (releases[index].get_version () == null) {
+                releases.remove_index (index);
+                index = 0;
```

Comment:
We have to go backwards to check the items that get pushed back by removing an item from the array, and we certainly don't want to go forwards in case that's now past the end of the array.

I've made it slightly more efficient and left a comment.